### DISCUSSION TITLE
Basic pytorch geometric architecture for graph signal classification

### DISCUSSION THREAD

**Discussion by Laolga** [pyg-team/pytorch_geometric#2407] [Q&A]
Hi! I'm very new to pytorch geometric, but my understanding would be that the most common applications are graph/node classifications while I'd be very interested in graph signal classification.
To clarify,  an example could be a feature matrix x (n,m) where for each author (n authors) we have indicators if they participated in a paper (m papers). Then we know the citation network for all papers (m by m adjacency matrix or (2,m) edge list. And we want to predict departments for the authors (so y is (1,n)). 
Of course, I could also reformulate the problem as a graph classification, but it does not look very memory efficient if every author uses exactly the same graph and this graph is very big (and possibly weighted).
I couldn't really find a similar setting in any of the examples. Please let me know if I missed something or there is an easy way to change one of the common examples to make it work in this setting.

**Comment by rusty1s** [comment]
To me, this looks like a node classification problem on a heterogeneous graph (you have a author<->paper graph and a paper<->paper graph). In fact, this is quite similar to the [`ogbn-mag` dataset](https://ogb.stanford.edu/). You can find some examples on how to handle such a heterogeneous setting [here](https://github.com/snap-stanford/ogb/tree/master/examples/nodeproppred/mag).

**Reply by Laolga** [reply]
Thanks a lot! This is indeed very helpful. 
Do you think it still makes sense to model the data as a heterogeneous graph if we assume that the abovementioned feature matrix is not space, but dense and very fat (m >> n)?

**Reply by rusty1s** [reply]
Personally, I would not interpret the mapping from authors to papers as a feature matrix, but instead as an additional edge connectivity, and rely on other reasonable node features (such as paper abstracts).

**Reply by Laolga** [reply]
Can this edge be weighted? I did not see so far a setting of R-GCN with weighted edges. Do you know any examples like that? Because the relation is not binary in my case, unfortunately, and not sparse either.
Thanks again for your help!

**Reply by rusty1s** [reply]
Edges can be weighted, and be processed by GNN ops such as `GraphConv` by passing in the `edge_weight` argument.
